src/Illuminate/Log/Context/Repository.php

```diff
@@ -0,0 +1,421 @@
+    {
+        return array_reduce($keys, function ($carry, $key) {
+            if (! $this->has($key)) {
+                return $carry;
+            }
+
+            return [
+                ...$carry,
+                ...[$key => $this->get($key)],
+            ];
+        }, []);
+    }
+
+    /**
+     * Retrieve only the secret values of the given keys.
+     *
+     * @param  array<int, string>  $keys
+     * @return array<string, mixed>
+     */
+    public function onlySecrets($keys)
+    {
+        return array_reduce($keys, function ($carry, $key) {
+            if (! $this->hasSecret($key)) {
+                return $carry;
+            }
+
+            return [
+                ...$carry,
+                ...[$key => $this->getSecret($key)],
+            ];
+        }, []);
+    }
+
+    /**
+     * Push the given values onto the key's stack.
+     *
+     * @param  string  $key
+     * @param  mixed  ...$values
+     * @return $this
+     */
+    public function push($key, ...$values)
+    {
+        if (! $this->isStackable($key)) {
+            throw new RuntimeException("Unable to push value onto context stack for key [{$key}].");
+        }
+
+        $this->data[$key] = [
+            ...$this->data[$key] ?? [],
+            ...$values,
+        ];
+
+        return $this;
+    }
+
+    /**
+     * Push the given values onto the key's secret stack.
+     *
+     * @param  string  $key
+     * @param  mixed  ...$values
+     * @return $this
+     */
+    public function pushSecret($key, ...$values)
+    {
+        if (! $this->isSecretStackable($key)) {
+            throw new RuntimeException("Unable to push value onto secret context stack for key [{$key}].");
+        }
+
+        $this->secrets[$key] = [
+            ...$this->secrets[$key] ?? [],
+            ...$values,
+        ];
+
+        return $this;
+    }
+
+    /**
+     * Determine if the given key exists.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    public function has($key)
+    {
+        return array_key_exists($key, $this->data);
+    }
+
+    /**
+     * Determine if the given key exists as a secret.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    public function hasSecret($key)
+    {
+        return array_key_exists($key, $this->secrets);
+    }
+
+    /**
+     * Remember the authenticated user.
+     *
+     * @return $this
+     */
+    public function rememberAuthUser()
+    {
+        $this->app['events']->listen([Login::class, Authenticated::class], fn (Login|Authenticated $event) => $this->addSecret('auth', [
+            'id' => $event->user->getAuthIdentifier(),
+            'guard' => $event->guard,
+        ]));
+
+        $this->app['events']->listen(fn (Logout $event) => $this->forgetSecret('auth'));
+
+        $this->app['events']->listen(function (Hydrated $event) {
+            $auth = $this->app['auth']->forgetGuards();
+
+            $this->restoreGuard ??= $auth->getDefaultDriver();
```

Comment:
It seems a bit concerning that when users opt-in to Remembering Auth User, they will automatically lose guards already loaded and have the context guard reprocessed or the default driver reloaded at a global level every time a new job starts being processed.

At least this is an opt-in functionality and developers can steer away from it.


a patch-release of Laravel (if this were to be merged) would go ahead and forget my Guards and reload the default auth driver without my consent

Reply:
Guards are no longer forgotten.